### PR TITLE
Add tracing collections examples and improve configuration

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,10 +53,199 @@ Myget feeds:
...
+// that flows in async calls.   When child is created, it
+// implicitly becomes child of current span
+using (tracer.WithSpan(parentSpan))
+{
+    var childSpan = tracer
+        .SpanBuilder("child")
+        .StartSpan();
+
+    childSpan.End();
+}
```
Comment (cijothomas) [inline README.md]: Who calls End() on parentSpan? Or is not neededed as its done automatically when using ends.?

Comment (lmolkova) [reply README.md]: dispose on Scope that WithSpan returns

File: README.md [hunk 2]
```diff
@@ -53,10 +53,199 @@ Myget feeds:
...
+var span = tracer
+    .SpanBuilder("span with attributes")
+    // spans have Client, Server, Internal, Producer and Consumer kinds to help visualize them
+    .SetSpanKind(SpanKind.Client)
+    .StartSpan();
+
+// attributes specific to the call
+span.SetAttribute("db.type", "redis");
+span.SetAttribute("db.instance", "localhost:6379[0]");
+span.SetAttribute("db.statement", "SET");
```
Comment (cijothomas) [inline README.md]: suggest to show variations which take non string values.
span.SetAttribute("db.Custom.IsLocalDb", true);

Comment (lmolkova) [reply README.md]: this is intended to show basic examples, not provide full documentation

File: README.md [hunk 3]
```diff
@@ -53,10 +53,199 @@ Myget feeds:
...
+span.End();
+```
+
+### Span with links
+
+Links allows to create relationships between different traces i.e. allow spans to have multiple relatives. They are typically used to tracer batching scenarios where multiple traces are merged into another one.
```
Comment (cijothomas) [inline README.md]: link to spec? (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#links-between-spans)

File: README.md [hunk 4]
```diff
@@ -53,10 +53,199 @@ Myget feeds:
...
+
+```csharp
+var span = tracer
+    .SpanBuilder("incoming HTTP request")
+    .SetSpanKind(SpanKind.Server)
+    .StartSpan();
+
+using (tracer.WithSpan(span))
+{
+    tracer.CurrentSpan.AddEvent("routes reolved");
```
Comment (cijothomas) [inline README.md]: example with attributes also.
var attr= new Dictionary<string, object>();
                attr.Add("key1", "val1");
                span.AddEvent("extra  event", attr);

Comment (lmolkova) [reply README.md]: I'd like to keep this doc short and not address advanced cases

**Pull request merged**